defmt: trait `Format` is not implemented for `Date`

using recent v0.16.1 i get error when using defmt 

```
   Compiling stm32f4xx-hal v0.16.1
error[E0277]: the trait bound `Date: Format` is not satisfied
   --> /home/alexxy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stm32f4xx-hal-0.16.1/src/rtc.rs:46:38
    |
46  | #[cfg_attr(feature = "defmt", derive(defmt::Format))]
    |                                      ^^^^^^^^^^^^^ the trait `Format` is not implemented for `Date`
    |
    = help: the following other types implement trait `Format`:
              &T
              &mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 172 others
note: required by a bound in `defmt::export::fmt`
   --> /home/alexxy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/defmt-0.3.5/src/export/mod.rs:137:15
    |
137 | pub fn fmt<T: Format + ?Sized>(f: &T) {
    |               ^^^^^^ required by this bound in `fmt`
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `Weekday: Format` is not satisfied
   --> /home/alexxy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/stm32f4xx-hal-0.16.1/src/rtc.rs:46:38
    |
46  | #[cfg_attr(feature = "defmt", derive(defmt::Format))]
    |                                      ^^^^^^^^^^^^^ the trait `Format` is not implemented for `Weekday`
    |
    = help: the following other types implement trait `Format`:
              &T
              &mut T
              ()
              (T0, T1)
              (T0, T1, T2)
              (T0, T1, T2, T3)
              (T0, T1, T2, T3, T4)
              (T0, T1, T2, T3, T4, T5)
            and 172 others
note: required by a bound in `defmt::export::fmt`
   --> /home/alexxy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/defmt-0.3.5/src/export/mod.rs:137:15
    |
137 | pub fn fmt<T: Format + ?Sized>(f: &T) {
    |               ^^^^^^ required by this bound in `fmt`
    = note: this error originates in the derive macro `defmt::Format` (in Nightly builds, run with -Z macro-backtrace for more info)


```

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -2,6 +2,7 @@
   "rust-analyzer.check.allTargets": false,
   "rust-analyzer.check.targets": "thumbv7em-none-eabihf",
   "rust-analyzer.cargo.features": [
+    "defmt",
     "rtic",
     "stm32f411"
   ],

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,15 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+### Changed
+
+  - enable `defmt` feature for VSCode
+  - `set_alarm` takes `Into<AlarmDay>`
+
+### Fixed
+
+  - Compilation with `defmt` feature enabled
+
 ## [v0.16.1] - 2023-06-24
 
  - bors bot replaced with GH merge queue

--- a/src/rtc.rs
+++ b/src/rtc.rs
@@ -43,12 +43,23 @@ impl From<Alarm> for Event {
     }
 }
 
-#[cfg_attr(feature = "defmt", derive(defmt::Format))]
 #[derive(Debug, Eq, PartialEq, Copy, Clone)]
 pub enum AlarmDay {
     Date(Date),
     Weekday(Weekday),
     EveryDay,
+}
+
+impl From<Date> for AlarmDay {
+    fn from(date: Date) -> Self {
+        Self::Date(date)
+    }
+}
+
+impl From<Weekday> for AlarmDay {
+    fn from(day: Weekday) -> Self {
+        Self::Weekday(day)
+    }
 }
 
 /// RTC clock source LSE oscillator clock (type state)
@@ -599,7 +610,13 @@ impl<CS> Rtc<CS> {
 
     /// Sets the time at which an alarm will be triggered
     /// This also clears the alarm flag if it is set
-    pub fn set_alarm(&mut self, alarm: Alarm, date: AlarmDay, time: Time) -> Result<(), Error> {
+    pub fn set_alarm(
+        &mut self,
+        alarm: Alarm,
+        date: impl Into<AlarmDay>,
+        time: Time,
+    ) -> Result<(), Error> {
+        let date = date.into();
         let (daymask, wdsel, (dt, du)) = match date {
             AlarmDay::Date(date) => (false, false, bcd2_encode(date.day().into())?),
             AlarmDay::Weekday(weekday) => (false, true, (0, weekday.number_days_from_monday())),

